package com.banking.common.logging;

import jakarta.servlet.FilterChain;
import jakarta.servlet.ServletException;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;
import org.springframework.web.filter.OncePerRequestFilter;

import java.io.IOException;

@Component
@Slf4j
public class LoggingFilter extends OncePerRequestFilter {

    @Override
    protected void doFilterInternal(

            HttpServletRequest request,

            HttpServletResponse response,

            FilterChain chain)

            throws ServletException, IOException {

        long start=System.currentTimeMillis();

        chain.doFilter(request,response);

        long end=System.currentTimeMillis();

        log.info(

                "{} {} {} ms",

                request.getMethod(),

                request.getRequestURI(),

                (end-start)

        );

    }

}
